crates/subspace-networking/src/protocols/connected_peers.rs

```diff
@@ -508,7 +529,9 @@ impl<Instance: 'static + Send> NetworkBehaviour for Behaviour<Instance> {
 
                     for (peer_id, address) in peer_addresses {
                         self.known_peers.entry(peer_id).or_insert_with(|| {
-                            ConnectionState::Connecting {
+                            cx.waker().wake_by_ref();
```

Comment:
I wonder, will it work while we process the current poll()?

Reply:
I would be VERY surprised if it didn't. This is how futures work and how this should work as well.